client/src/controllers/RevealController.ts

```diff
@@ -0,0 +1,311 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Adds the ability to make the controlled element be used as an
+ * opening/closing (aka collapsing) element.
+ * Supports, and relies on, correct usage of aria-* attributes.
```

Comment:
With this description I think it's more fitting if we name it something like `ExpandController` or `CollapsibleController`. The former seems short and expressive enough, but not sure if that would cause any confusion with `aria-expand`. I don't think so, and it makes sense (at least to me) that this would control `aria-expand` just because it's clearly controlling an element that can be expanded/collapsed.

Reply:
I started this way when first looking into this but ran into a few issues.

1. Common typos - Collapsible vs Collapsable - https://thecontentauthority.com/blog/collapsable-vs-collapsible 
2. Longer name makes all attributes longer
3. It makes it tricky to NOT use the 'state' of expanded or collapsed and the 'actions' of expand or collapse. Using hide/show is not the most suitable here and is better for things that fully hide/show (e.g. tooltips, dialog use hide/show).

Additionally, based on feedback on the Panel issue https://github.com/wagtail/wagtail/issues/10168#issuecomment-1483563649 we found that there is already a nice precedent in the community for naming this https://www.stimulus-components.com/docs/stimulus-reveal-controller/

Our existing implementations have the following

* https://github.com/wagtail/wagtail/blob/main/client/src/includes/breadcrumbs.js (open/closed)
* https://github.com/wagtail/wagtail/blob/main/client/src/entrypoints/admin/preview-panel.js (show/hide)
* https://github.com/wagtail/wagtail/blob/main/client/src/includes/panels.ts (expand/collapse)

If it's critical I am more than happy to re-write this using the Expand/Collapse naming but I did go down this way already and it felt more verbose.